analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TooManyParameters_CustomValues.CSharp11.cs

```diff
@@ -17,4 +17,46 @@ public interface If
         static abstract void F1(int p1, int p2, int p3);
         static abstract void F2(int p1, int p2, int p3, int p4); // Noncompliant
     }
+
+    public unsafe partial class LibraryImportAttributeImports
+    {
+        [LibraryImport("foo.dll")]
+        public static partial void ExternWithoutMarshaling(int p1, int p2, int p3, int p4); // Compliant, external definition
+
+        // Provide the implementation usually generated by the source generator
+        [System.Runtime.InteropServices.DllImportAttribute("foo.dll", EntryPoint = "Extern", ExactSpelling = true)]
+        public static extern partial void ExternWithoutMarshaling(int p1, int p2, int p3, int p4);
+
+        [LibraryImport("nativelib", EntryPoint = "ExternWithMarshaling", StringMarshalling = StringMarshalling.Utf16)]
+        internal static partial string ExternWithMarshaling(string str, int p1, int p2, int p3, int p4); // Compliant, external definition
+
+        // Provide the implementation usually generated by the source generator.
+        [System.Runtime.CompilerServices.SkipLocalsInitAttribute]
+        internal static partial string ExternWithMarshaling(string str, int p1, int p2, int p3, int p4)
+        {
```

Comment:
is the method implementation really necessary for testing this rule?

Reply:
I'm mainly wondering whether this should be in an integration test, that ensures the code is compiled

We have a project for this: https://github.com/SonarSource/sonar-dotnet/tree/master/analyzers/its/sources/ManuallyAddedNoncompliantIssues.CS/IntentionalFindings

And there you could also put a README or a more extensive comment on how to regenerate the code using the Source Generator, for example (the step-by-step approach), or even add the project that is necessary to do it.

Reply:
I added code comments in the UT about how to re-create the generated code because that is the place where it is needed.